src/freertos_drivers/spiffs/cc32x0sf/CC32x0SFSPIFFS.cxx

```diff
@@ -31,11 +31,64 @@
  * @date 1 January 2018
  */
 
+// #define LOGLEVEL INFO
+
+// This define is needed to call any ROM_xx function in the driverlib.
+#define USE_CC3220_ROM_DRV_API
+
+#include "utils/logging.h"
+
 #include "CC32x0SFSPIFFS.hxx"
 
 #include "spiffs.h"
 #include "inc/hw_types.h"
 #include "driverlib/flash.h"
+#include "driverlib/rom.h"
+#include "driverlib/cpu.h"
+
+/// Flash configuration register.
+#define FLASH_CONF 0x400FDFC8
+/// This bit in the FLASH_CONF register means that the banks are reversed by
+/// their address mapping.
+#define FCMME 0x40000000
+/// This value defines up to one bit that needs to be XOR-ed to the flash
+/// address before calling the flash APIs to cover for reversed flash banks.
+static const unsigned addr_mirror = (HWREG(FLASH_CONF) & FCMME) ? 0x80000 : 0;
+
+// Different options for what to set for flash write locking.
+
+// Global disable interrupts.
```

Comment:
How is the option chosen at an application specific level? Would it be better to define these as weak functions that the application can override?

Reply:
It is not expected that the application will have to choose, but these are alternatives for debugging only.